middleware/applyOverrides.js

```diff
@@ -0,0 +1,41 @@
+const _ = require('lodash');
+const logger = require('pelias-logger').get('api');
+const codec = require('pelias-model').codec;
+
+function setup() {
+  return applyOverrides;
+}
+
+function applyOverrides(req, res, next) {
+  // do nothing if no result data set
+  if (!res || !res.data) {
+    return next();
+  }
+
+  res.data = res.data.map(overrideOneRecord);
+
+  next();
+}
+
+/*
+ * Rename the fields in one record
+ */
+function overrideOneRecord(place) {
+  if (place.addendum && place.addendum.override) {
+    try {
+      const overrideData = codec.decode(place.addendum.override);
+      place = _.merge(place, overrideData);
+    } catch (err) {
+      logger.error('Invalid addendum override json string:', place.addendum);
+    }
+
+    delete place.addendum.override;
+
+    if (_.isEmpty(place.addendum)) {
+      delete place.addendum;
+    }
```

Comment:
is it possible to use `_.unset()` for this or does it not remove the empty parent object?

Reply:
wouldn't remove empty parent